SeamCarver.js

```diff
@@ -86,22 +94,28 @@ class SeamCarver {
      * @return {number} The energy val.
      */
     energy(x, y) {
-        if (x <= 0 || y <= 0 || x >= this.width-1 || y >= this.height-1) {
+        if (this.isBorderPixel(x, y)) {
             return BORDER_ENERGY;
         }
-        var xant = this.context.getImageData(x - 1, y, 1, 1).data;
-        var xpost = this.context.getImageData(x + 1, y, 1, 1).data;
-        var yant = this.context.getImageData(x, y - 1, 1, 1).data;
-        var ypost = this.context.getImageData(x, y + 1, 1, 1).data;
 
-        return Math.sqrt(
+        var pos = this.pixelToIndex(x - 1, y);
+        var xant  = Array.prototype.slice.call(this.picture, pos, pos + 3);
```

Comment:
These `slice` calls really do make things slow. I profiled the demo. It's not iterating the array, just index adjustment. I'll make a pull request for this.